jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -817,6 +817,57 @@ ecma_builtin_helper_def_prop_by_index (ecma_object_t *obj_p, /**< object */
   return ret_value;
 } /* ecma_builtin_helper_def_prop_by_index */
 
+/**
+ * Helper function for at() functions.
+ *
+ * See also:
+ *          ECMA-262 Stage 3 Draft Relative Indexing Method proposal 3. 4. 5. 6.
+ *
+ * Used by:
+ *         - The Array.prototype.at routine.
+ *         - The String.prototype.at routine.
+ *         - The TypedArray.prototype.at routine.
+ *
+ * @return ECMA_VALUE_ERROR - on conversion error
+ *         ECMA_VALUE_UNDEFINED - if the requested index is not exist
+ *         ECMA_VALUE_EMPTY - otherwise
+ */
+ecma_value_t
+ecma_builtin_helper_check_index (ecma_value_t index, /**< index argument */
+                                 ecma_number_t *len) /**< [out] object's length */
```

Comment:
Is this really the length? It seems to me that this is used as input length, but it is also used to return the calculated index. Which is weird. It should be better to have a third argument which returns the calculated index, eg.:
```c
ecma_builtin_helper_check_index(ecma_value_t index,
                                ecma_length_t length,
                                ecma_number_t *out_index)
```
This way the input length can be of `ecma_length_t` type.

Also:
* Should we return `ecma_number_t` at all? I think we could return `ecma_length_t` typed value. (This needs a check on how there resulting index is used in most cases).
* This method does not "check" but "calulate"-s the required index. Maybe we could use the `ecma_builtin_helper_calculate_index` name?

Reply:
It could be works with ecma_length_t. Im not sure which one is better in here , ecma_length_t or ecma_number_t. (Recent patch)